cou/steps/hypervisor.py

```diff
@@ -261,7 +265,12 @@ def _generate_post_upgrade_steps(
 
             units = group.app_units[app.name]
             logger.info("generating post-upgrade steps for %s units of %s app", app.name, units)
-            steps.extend(app.post_upgrade_steps(target, units=units))
+            if app.charm == "nova-compute":
+                # NOTE(gabrielcocenza) it's important to enable the scheduler after upgrading
+                # all applications.
+                steps.extend(app.post_upgrade_steps(target, units=units))
+            else:
+                steps = app.post_upgrade_steps(target, units) + steps
```

Comment:
Same here.

Reply:
As said above, the app sequence might start with OpenStack apps if colocatd. This probably it's not necessary, but it makes it explicit.

Reply:
Since post-upgrade are only there to chceck and not upgrade, is the order when scheduler is enabled important?